caches/infinispan/src/main/java/org/commonjava/maven/galley/cache/infinispan/FastLocalCacheProviderFactory.java

```diff
@@ -51,6 +51,10 @@ public FastLocalCacheProviderFactory( File cacheDir, File nfsDir, Cache<String,
         this.executor = executor;
     }
 
+    public FastLocalCacheProviderFactory(File cacheDir, File nfsDir, ExecutorService executor){
+        this(cacheDir, nfsDir, null, executor);
```

Comment:
Is there a reason to support this? I suppose it makes it easier to embed, but my concern is that it may lead to poor performance characteristics if we create a cache instance on-the-fly.